sgrep.py

```diff
@@ -270,27 +270,37 @@ def rewrite_message_with_metavars(yaml_rule, sgrep_result):
     return msg_text
 
 
-def collect_rules(yaml_file_or_dirs: str) -> Tuple[List[Dict[str, Any]], Tuple[int, int]]:
+def _collect_rules_from_files(files: List[str], root: Optional[str] = None)-> Tuple[List[Dict[str, Any]], Tuple[int, int]]:
     collected_rules = []
     errors, not_errors = 0, 0
-    for root, dirs, files in os.walk(yaml_file_or_dirs):
-        dirs.sort()
-        for filename in sorted(files):
-            if pathlib.Path(filename).suffix in YML_EXTENSIONS:
+    for filename in sorted(files):
+        if pathlib.Path(filename).suffix in YML_EXTENSIONS:
+            if root:
                 full_path = os.path.join(root, filename)
-                rules_in_file = parse_sgrep_yml(full_path)
-                if rules_in_file is None:
-                    errors += 1
-                else:
-                    not_errors += 1
-                    for rule in rules_in_file:
-                        prefix = '.'.join([x for x in PurePath(
-                            pathlib.Path(full_path)).parts[:-1] if len(x)])
-                        new_id = f"{prefix}.{rule['id']}".lstrip('.')
-                        rule['id'] = new_id
-                    collected_rules.extend(rules_in_file)
+            else:
+                full_path = filename
+            rules_in_file = parse_sgrep_yml(full_path)
+            if rules_in_file is None:
+                errors += 1
+            else:
+                not_errors += 1
+                for rule in rules_in_file:
+                    prefix = '.'.join([x for x in PurePath(
+                        pathlib.Path(full_path)).parts[:-1] if len(x)])
+                    new_id = f"{prefix}.{rule['id']}".lstrip('.')
+                    rule['id'] = new_id
+                collected_rules.extend(rules_in_file)
     return collected_rules, (errors, not_errors)
 
+def collect_rules(yaml_file_or_dirs: str) -> Tuple[List[Dict[str, Any]], Tuple[int, int]]:
+    if os.path.isfile(yaml_file_or_dirs):
+        file_path = os.path.abspath(yaml_file_or_dirs)
+        return _collect_rules_from_files([file_path])
+
+    for root, dirs, files in os.walk(yaml_file_or_dirs):
```

Comment:
@ulziibay should this be under an `else:`

Reply:
Same logic no? This version is better because of better indentation